### PR TITLE
Bug 1303036 - Use treeherder's job details API instead of a privatebuild artifact;r=bc

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

File: autophonetreeherder.py [hunk 1]
```diff
@@ -191,13 +191,23 @@ def _create_job(self, tjc, machine, build_url, project, revision, build_type, bu
...
+        privatebuild = {
             'build_url': build_url,
             'config_file': t.config_file,
             'chunk': t.chunk,
-            'builder_type': builder_type})
+            'builder_type': builder_type
+        }
+        tj.add_artifact('Job Info', 'json', {
+            'job_details': [
+                {'title': k, 'value': v } for
```
Comment (bclary) [inline autophonetreeherder.py]: C:206, 0: No space allowed before bracket
                {'title': k, 'value': v } for
                                        ^ (bad-whitespace)

File: autophonepulsemonitor.py [hunk 2]
```diff
@@ -521,14 +518,25 @@ def get_treeherder_job(self, project, job_id):
...
-                if artifact['name'] == 'privatebuild':
-                    url = '%s%s' % (
-                        self.treeherder_url, artifact['resource_uri'])
-                    return utils.get_remote_json(url)
-        return None
+    def get_treeherder_privatebuild_info(self, project, job):
+        url = '%s%s' % (
+            self.treeherder_url,
+            '/jobdetail/?repository=%s&job_id=%s' % (
+                project, job['id')))
```
Comment (bclary) [inline autophonepulsemonitor.py]: I don't get the benefit of this nested string interpolation. Can we do it all at once like 

        url = '%s/jobdetail/?repository=%s&job_id=%s' % (
            self.treeherder_url, project, job['id'])

Comment (wlach) [inline autophonepulsemonitor.py]: Makes sense, will fix.

File: autophonepulsemonitor.py [hunk 3]
```diff
@@ -521,14 +518,28 @@ def get_treeherder_job(self, project, job_id):
...
+                # the build_url property is too long to fit into "value"
+                if detail.get('url'):
+                    value = detail['url']
+                else:
+                    value = detail['value']
+                info[detail['title']] = value
+
+        # if we're missing a privatebuild key for some reason
+        # return None to avoid errors
+        if set(info.keys()) != privatebuild_keys:
```
Comment (bclary) [inline autophonepulsemonitor.py]: perhaps LOGGER.warning the missing keys?

**Review by bclary — CHANGES_REQUESTED** [review]
Works for me as is. I'd like a warning as mentioned.

File: autophonepulsemonitor.py (hunk 3 above)
Comment (wlach) [inline autophonepulsemonitor.py]: Sure :)

**Comment by bclary** [comment]
This project is no longer active and is being archived.

**Pull request closed without merging**